src/NUnitFramework/framework/Internal/Reflect.cs

```diff
@@ -78,6 +78,35 @@ public static object Construct(Type type, object?[]? arguments)
             ConstructorInfo? ctor = GetConstructors(type, argTypes).FirstOrDefault();
             if (ctor is null)
                 throw new InvalidTestFixtureException(type.FullName + " does not have a suitable constructor");
+            ParameterInfo[] parameterInfos = ctor.GetParameters();
+
+            if (parameterInfos.Length > 0)
+            {
+                ParameterInfo parameterInfo = parameterInfos.Last();
+                if (parameterInfo.HasAttribute<ParamArrayAttribute>(false))
+                {
+                    if (arguments.Length == parameterInfos.Length 
+                        && parameterInfo.ParameterType.IsAssignableFrom(arguments[parameterInfos.Length - 1]?.GetType()))
+                    {
+                        // Don't convert arguments as there was already an array we could use.
+                    }
+                    else
+                    {
+                        Type? elementType = parameterInfo.ParameterType.GetElementType();
+                        if (elementType is null)
+                        {
+                            throw new InvalidTestFixtureException(type.FullName + " params argument did not have an element type");
+                        }
+                        int paramsOffset = ctor.GetParameters().Length - 1;
+                        var paramArray = Array.CreateInstance(elementType, argTypes.Length - ctor.GetParameters().Length + 1);
```

Comment:
You are calling the `ctor.GetParameters()` twice again, just to get its length. However the return value of this method was previously assigned to `parametersInfos`. So please replace the function calls with references to the variable.